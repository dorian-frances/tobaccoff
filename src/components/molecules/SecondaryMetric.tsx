import React from 'react';
import { View } from 'react-native';
import MetricText, { MetricTextProps } from '../atoms/texts/MetricText';
import SectionText, { SectionTextProps } from '../atoms/texts/SectionText';

export type SecondaryMetricProps = {
  metricTextProps: MetricTextProps;
  sectionTextProps: SectionTextProps;
};

const SecondaryMetric = ({
  metricTextProps,
  sectionTextProps,
}: SecondaryMetricProps) => {
  return (
    <View style={{ alignItems: 'center' }}>
      <MetricText {...metricTextProps} />
      <SectionText {...sectionTextProps} />
    </View>
  );
};

export default SecondaryMetric;
