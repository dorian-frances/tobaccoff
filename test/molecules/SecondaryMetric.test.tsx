import React from 'react';
import { render } from '@testing-library/react-native';
import SecondaryMetric from '../../src/components/molecules/SecondaryMetric';

describe('<SmokedCigarettes/>', () => {
  it('should render correctly', () => {
    const component = render(
      <SecondaryMetric
        sectionTextProps={{ text: 'Cigarettes fumÃ©es', fontSize: 12 }}
        metricTextProps={{
          metric: 50,
          unit: '',
          minimumFractionDigits: 0,
          fontSize: 20,
        }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
