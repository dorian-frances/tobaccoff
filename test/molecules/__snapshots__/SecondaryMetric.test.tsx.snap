// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmokedCigarettes/> should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 20,
        },
      ]
    }
  >
    50
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Montserrat-Medium",
        "fontSize": 12,
      }
    }
  >
    Cigarettes fum√©es
  </Text>
</View>
`;
